### PR TITLE
 Add Trino 469 release notes

### PR DESCRIPTION
**Pull request by mosabua**
## Description

Assemble the release notes for the upcoming Trino 469 release.

## Additional context and related issues

See https://github.com/trinodb/trino/pulls?q=is%3Apr+is%3Aclosed+milestone%3A469

## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.

## Verification for each pull request

Format: PR/issue number, ✅ / ❌ rn ✅ / ❌ docs
✅ rn - release note added and verified, or assessed to be not necessary, set to ❌ rn before completion
✅ docs - need for docs assessed and merged, or assessed to be not necessary, set to ❌ docs before completion

## 18 Dec 2024

* #24507 ✅ rn ✅ docs
* #24447 ✅ rn ✅ docs
* #24514 ✅ rn ✅ docs
* #24520 ✅ rn ✅ docs
* #24418 ✅ rn ✅ docs
* #24501 ✅ rn ✅ docs

## 19 Dec 2024

* #24529 ✅ rn ✅ docs
* #24527 ✅ rn ✅ docs
* #24508 ✅ rn ✅ docs
* #24521 ✅ rn ✅ docs
* #24532 ✅ rn ✅ docs
* #24536 ✅ rn ✅ docs
* #24524 ✅ rn ✅ docs
* #24504 ✅ rn ✅ docs
* #24172 ✅ rn ✅ docs

## 20 Dec 2024

* #24548 ✅ rn ✅ docs
* #24550 ✅ rn ✅ docs

## 21 Dec 2024

* #24554 ✅ rn ✅ docs

## 22 Dec 2024

* #23909 ✅ rn ✅ docs
* #24553 ✅ rn ✅ docs

## 23 Dec 2024

* #24560 ✅ rn ✅ docs
* #24469 ✅ rn ✅ docs, reverted
* #24443 ✅ rn ✅ docs, #24523
* #24564 ✅ rn ✅ docs
* #24543 ✅ rn ✅ docs, #24542 and #24686
* #24306 ✅ rn ✅ docs

## 24 Dec 2024

* #24557 ✅ rn ✅ docs
* #24491 ✅ rn ✅ docs
* #17426 ✅ rn ✅ docs
* #24571 ✅ rn ✅ docs
* #24573 ✅ rn ✅ docs

## 25 Dec 2024

* #24574 ✅ rn ✅ docs
* #24242 ✅ rn ✅ docs

## 26 Dec 2024

* #24581 ✅ rn ✅ docs
* #24323 ✅ rn ✅ docs
* #24583 ✅ rn ✅ docs

## 27 Dec 2024

* #24589 ✅ rn ✅ docs

## 28 Dec 2024

* #24578 ✅ rn ✅ docs
* #24337 ✅ rn ✅ docs

## 29 Dec 2024

No PRs merged.

## 30 Dec 2024

* #24533 ✅ rn ✅ docs
* #24594 ✅ rn ✅ docs
* #24405 ✅ rn ✅ docs

## 31 Dec 2024

* #24590 ✅ rn ✅ docs
* #24595 ✅ rn ✅ docs
* #24531 ✅ rn ✅ docs
* #24602 ✅ rn ✅ docs
* #24603 ✅ rn ✅ docs
* #24117 ✅ rn ✅ docs
* #24605 ✅ rn ✅ docs
* #24604 ✅ rn ✅ docs

## 1 Jan 2025

* #23815 ✅ rn ✅ docs
* #24607 ✅ rn ✅ docs
* #24608 ✅ rn ✅ docs
* #24523 ✅ rn ✅ docs
* #22899 ✅ rn ✅ docs
* #24566 ✅ rn ✅ docs
* #24609 ✅ rn ✅ docs
* #24610 ✅ rn ✅ docs

## 2 Jan 2025

* #24228 ✅ rn ✅ docs
* #24606 ✅ rn ✅ docs
* #24613 ✅ rn ✅ docs

## 3 Jan 2025

* #24618 ✅ rn ✅ docs

## 4 Jan 2025

No PRs merged.

## 5 Jan 2025

* #24614 ✅ rn ✅ docs
* #24630 ✅ rn ✅ docs

## 6 Jan 2025

* #24622 ✅ rn ✅ docs
* #24626 ✅ rn ✅ docs
* #24634 ✅ rn ✅ docs
* #24584 ✅ rn ✅ docs

## 7 Jan 2025

* #24623 ✅ rn ✅ docs
* #24580 ✅ rn ✅ docs
* #24640 ✅ rn ✅ docs
* #24639 ✅ rn ✅ docs
* #24612 ✅ rn ✅ docs
* #24644 ✅ rn ✅ docs
* #24645 ✅ rn ✅ docs
* #24519 ✅ rn ✅ docs
* #24478 ✅ rn ✅ docs
* #24438 ✅ rn ✅ docs
* #24410 ✅ rn ✅ docs

## 8 Jan 2025

* #24649 ✅ rn ✅ docs
* #24632 ✅ rn ✅ docs
* #24653 ✅ rn ✅ docs
* #24633 ✅ rn ✅ docs
* #24619 ✅ rn ✅ docs
* #24656 ✅ rn ✅ docs
* #24657 ✅ rn ✅ docs
* #24635 ✅ rn ✅ docs
* #24655 ✅ rn ✅ docs, #24670

## 9 Jan 2025

* #24664 ✅ rn ✅ docs
* #24665 ✅ rn ✅ docs
* #24666 ✅ rn ✅ docs
* #24670 ✅ rn ✅ docs
* #24542 ✅ rn ✅ docs
* #24285 ✅ rn ✅ docs
* #24669 ✅ rn ✅ docs
* #24672 ✅ rn ✅ docs
* #24671 ✅ rn ✅ docs

## 10 Jan 2025

* #24677 ✅ rn ✅ docs
* #24297 ✅ rn ✅ docs
* #24651 ✅ rn ✅ docs
* #23230 ✅ rn ✅ docs
* #24667 ✅ rn ✅ docs
* #24681 ✅ rn ✅ docs
* #24681 ✅ rn ✅ docs
* #24515 ✅ rn ✅ docs
* #24682 ✅ rn ✅ docs
* #24683 ✅ rn ✅ docs
* #24127 ✅ rn ✅ docs
* #24373 ✅ rn ✅ docs
* #23544 ✅ rn ✅ docs
* #20152 ✅ rn ✅ docs
* #24673 ✅ rn ✅ docs

## 11 Jan 2025

* #24475 ✅ rn ✅ docs
* #24686 ✅ rn ✅ docs

## 12 Jan 2025

* #24691 ✅ rn ✅ docs
* #24674 ✅ rn ✅ docs

## 13 Jan 2025

* #24693 ✅ rn ✅ docs
* #24688 ✅ rn ✅ docs
* #24694 ✅ rn ✅ docs
* #21520 ✅ rn ✅ docs
* #24558 ✅ rn ✅ docs
* #24690 ✅ rn ✅ docs
* #24638 ✅ rn ✅ docs

## 14 Jan 2025

* #22443 ✅ rn ✅ docs
* #24697 ✅ rn ✅ docs
* #24706 ✅ rn ✅ docs
* #24494 ✅ rn ✅ docs
* #24689 ✅ rn ✅ docs

## 15 Jan 2025

* #24711 ✅ rn ✅ docs
* #24700 ✅ rn ✅ docs
* #24712 ✅ rn ✅ docs
* #24710 ✅ rn ✅ docs
* #24719 ✅ rn ✅ docs

## 16 Jan 2025

* #24244 ✅ rn ✅ docs
* #24723 ✅ rn ✅ docs
* #24721 ✅ rn ✅ docs
* #24727 ✅ rn ✅ docs
* #20914 ✅ rn ✅ docs

## 17 Jan 2025

* #23761 ✅ rn ✅ docs
* #23839 ✅ rn ✅ docs
* #24736 ✅ rn ✅ docs

## 18 Jan 2025

No PRs merged.

## 19 Jan 2025

* #24743 ✅ rn ✅ docs

## 20 Jan 2025

* #24704 ✅ rn ✅ docs
* #24742 ✅ rn ✅ docs
* #24747 ✅ rn ✅ docs
* #24740 ✅ rn ✅ docs
* #24750 ✅ rn ✅ docs
* #24752 ✅ rn ✅ docs

## 21 Jan 2025

* #24733 ✅ rn ✅ docs
* #24757 ✅ rn ✅ docs, see #24761

## 22 Jan 2025

* #24762 ✅ rn ✅ docs
* #24766 ✅ rn ✅ docs
* #24764 ✅ rn ✅ docs
* #24763 ✅ rn ✅ docs
* #24769 ✅ rn ✅ docs

## 23 Jan 2025

* #24777 ✅ rn ✅ docs
* #24709 ✅ rn ✅ docs
* #24781 ✅ rn ✅ docs
* #24774 ✅ rn ✅ docs
* #24748 ✅ rn ✅ docs
* #24761 ✅ rn ✅ docs
* #24778 ✅ rn ✅ docs
* #24735 ✅ rn ✅ docs

## 24 Jan 2025

No PRs merged.

## 25 Jan 2025

No PRs merged.

## 26 Jan 2025

No PRs merged.

## 27 Jan 2025

* #24803 ✅ rn ✅ docs
* #24787 ✅ rn ✅ docs
* #20457 ✅ rn ✅ docs

### CODE REVIEW

File: docs/src/main/sphinx/release/release-469.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+## Blackhole connector
+
+## Cassandra connector
+
+## ClickHouse connector
+
+## Delta Lake connector
+
+* Improve performance of reading from Delta Lake tables by compressing files
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-469.md:29]: @raunaqmorarka .. how does this work with existing stuff. I assume the files are unmodified as usual so it only applies to new data .. so reading is NOT accelerated for old data .. only for newly written files that are subsequently read .. and as such.. do we need to reword @martint .. also note I added the info about SNAPPY being the old default.

File: docs/src/main/sphinx/release/release-469.md [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+## Druid connector
+
+## Elasticsearch connector
+
+## Exasol connector
+
+## Faker connector
+
+* Add support for views. ({issue}`24242`)
+* Replace predicate pushdown with `min`, `max`, and `options` column properties. ({issue}`24147`)
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-469.md:44]: I think we probaly want to just remove the mention of predicate pushdown and mention addition of the column properties and what they add.

File: docs/src/main/sphinx/release/release-469.md [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+## Druid connector
+
+## Elasticsearch connector
+
+## Exasol connector
+
+## Faker connector
+
+* Add support for views. ({issue}`24242`)
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-469.md:43]: Thanks for adding the docs entry for the SQL support section. Question though .. do we need to add more specific info like .. where are those views stored? Do they survive a cluster restart?

Comment (mosabua) [reply docs/src/main/sphinx/release/release-469.md:43]: And are create, drop and alter supported?

File: docs/src/main/sphinx/release/release-469.md (hunk 2 above)
Comment (nineinchnick) [reply docs/src/main/sphinx/release/release-469.md:44]: It's breaking change, so we should definitely say that predicate pushdown are not supported anymore.

File: docs/src/main/sphinx/release/release-469.md (hunk 3 above)
Comment (nineinchnick) [reply docs/src/main/sphinx/release/release-469.md:43]: The implementation is copied from the memory connector. The views are stored in memory, same as the tables are, and we only mention this for tables. There are tests that prove we support CREATE VIEW, DROP VIEW and ALTER VIEW statements.

File: docs/src/main/sphinx/release/release-469.md (hunk 1 above)
Comment (raunaqmorarka) [reply docs/src/main/sphinx/release/release-469.md:34]: There will be no change existing files, newly written files will use zstd

File: docs/src/main/sphinx/release/release-469.md [hunk 4]
```diff
@@ -0,0 +1,125 @@
+# Release 469 (8 Jan 2025)
+
+## General
+
+* Refrain from join pushdown for modified tables. ({issue}`24447`)
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-469.md:5]: I think we need to reword this @martint

File: docs/src/main/sphinx/release/release-469.md [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+* Fix parsing of negative `0x`, `0b`, `0o` long literals. ({issue}`24601`)
+
+## Security
+
+## Web UI
+
+## JDBC driver
+
+* Add `physicalInputTimeMillis` to `io.trino.jdbc.QueryStats`. ({issue}`24604`)
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-469.md:14]: Do we want this in the release notes at all.. I think not @martint

File: docs/src/main/sphinx/release/release-469.md [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+
+## Blackhole connector
+
+## Cassandra connector
+
+## ClickHouse connector
+
+## Delta Lake connector
+
+* Improve performance of reading from new Delta Lake tables by compressing files
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-469.md:35]: fyi .. a bit reworded @raunaqmorarka

File: docs/src/main/sphinx/release/release-469.md [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+* Support generating sequences. ({issue}`24590`)
+* {{breaking}} Replace predicate pushdown with `min`, `max`, and `options`
+  column properties. ({issue}`24147`)
+
+## Google Sheets connector
+
+## Hive connector
+
+* Fix JMX metrics for file system caching with multiple catalogs. ({issue}`24510`)
+* Fix reading from tables when using the Alluxio file system. ({issue}`23815`)
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-469.md:59]: @raunaqmorarka is this really only for the Hive connector .. or also Iceberg, Delta Lake, Hudi?

File: docs/src/main/sphinx/release/release-469.md [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+## Hudi connector
+
+## Iceberg connector
+
+* Add `$entries` metadata table. ({issue}`24172`)
+* Add `$all_entries` metadata table. ({issue}`24543`)
+* Add `system.iceberg_tables` table function to allow listing only Iceberg tables. ({issue}`24469`)
+* Allow configuring the `parquet_bloom_filter_columns` table property. ({issue}`24573`)
+* Improve performance when listing columns. ({issue}`23909`)
+* Remove the oldest tracked version metadata files when
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-469.md:70]: Is this a bug fix? How can we better describe the user impact @ebyhr ? 

From quick glance it seems a clean up work that has performance benefits..

File: docs/src/main/sphinx/release/release-469.md (hunk 7 above)
Comment (raunaqmorarka) [reply docs/src/main/sphinx/release/release-469.md:59]: Iceberg and delta as well

File: docs/src/main/sphinx/release/release-469.md (hunk 6 above)
Comment (raunaqmorarka) [reply docs/src/main/sphinx/release/release-469.md:35]: It applies to new files in existing tables as well since compression is not a table property

File: docs/src/main/sphinx/release/release-469.md (hunk 5 above)
Comment (raunaqmorarka) [reply docs/src/main/sphinx/release/release-469.md:14]: This is something exposed for users of JDBC driver

Comment (mosabua) [reply docs/src/main/sphinx/release/release-469.md:14]: ok .. will leave then.

File: docs/src/main/sphinx/release/release-469.md (hunk 7 above)
Comment (mosabua) [reply docs/src/main/sphinx/release/release-469.md:59]: thx

File: docs/src/main/sphinx/release/release-469.md [hunk 9]
```diff
@@ -0,0 +1,143 @@
...
+
+## TPC-H connector
+
+## TPC-DS connector
+
+## Vertica connector
+
+## SPI
+
+* Remove `Connector.getEventListeners()`. ({issue}`24609`)
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-469.md:143]: Should we reword this to 

```
Remove support for connector-level event listeners and the related `Connector.getEventListeners()` method.
```

Comment (mosabua) [reply docs/src/main/sphinx/release/release-469.md]: Did that now

File: docs/src/main/sphinx/release/release-469.md [hunk 10]
```diff
@@ -0,0 +1,136 @@
+# Release 469 (14 Jan 2025)
+
+## General
+
+* Add geospatial constructor {func}`ST_GeomFromKML`. ({issue}`24297`)
+* Refrain from join pushdown for modified tables. ({issue}`24447`)
+* Allow deactivating the spooling protocol with the `spooling_protocol_enabled`
+  session property. ({issue}`24655`)
+* Fix coordinator out of memory issues caused by a Jetty upgrade in Trino 468. ({issue}`24572`)
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-469.md:9]: Not sure if we refer to version of regression and cause typically .. happy to remove or reword. Thoughts @wendigo @martint ?

Comment (wendigo) [reply docs/src/main/sphinx/release/release-469.md:9]: It's both coordinator and worker. I'd just say "improve stability of the cluster under load" ?

File: docs/src/main/sphinx/release/release-469.md [hunk 11]
```diff
@@ -0,0 +1,136 @@
...
+## JDBC driver
+
+* Add `planningTimeMillis`, `analysisTimeMillis`, `finishingTimeMillis`,
+  `physicalInputBytes`, `physicalWrittenBytes`, `internalNetworkInputBytes` and
+  `physicalInputTimeMillis` to `io.trino.jdbc.QueryStats`. ({issue}`24571`,
+  {issue}`24604`)
+* Add the `validateConnection` connection property to allows validating
+  connection. ({issue}`16504`)
+* Fix loading spooled segment data when cluster is secured. ({issue}`24595`)
+* Fix deserialization of `SetDigest`, `BingTile`, and `Color` types. ({issue}`24612`)
```
[the remark this answers is not in the document]
Comment (wendigo) [reply docs/src/main/sphinx/release/release-469.md:29]: Deserialization failure

File: docs/src/main/sphinx/release/release-469.md [hunk 12]
```diff
@@ -0,0 +1,136 @@
...
+  descending reserved memory. ({issue}`24475`)
+
+## JDBC driver
+
+* Add `planningTimeMillis`, `analysisTimeMillis`, `finishingTimeMillis`,
+  `physicalInputBytes`, `physicalWrittenBytes`, `internalNetworkInputBytes` and
+  `physicalInputTimeMillis` to `io.trino.jdbc.QueryStats`. ({issue}`24571`,
+  {issue}`24604`)
+* Add the `validateConnection` connection property to allows validating
+  connection. ({issue}`16504`)
```
[the remark this answers is not in the document]
Comment (mosabua) [reply docs/src/main/sphinx/release/release-469.md:27]: Done

File: docs/src/main/sphinx/release/release-469.md [hunk 13]
```diff
@@ -0,0 +1,136 @@
...
+* Fix loading spooled segment data when cluster is secured. ({issue}`24595`)
+* Fix deserialization of `SetDigest`, `BingTile`, and `Color` types. ({issue}`24612`)
+* Fix `Connection.isValid(int)` so it validates connection and credentials. ({issue}`22684`)
```
[the remark this answers is not in the document]
Comment (mosabua) [reply docs/src/main/sphinx/release/release-469.md:30]: Makes sense.. done

File: docs/src/main/sphinx/release/release-469.md [hunk 14]
```diff
@@ -0,0 +1,136 @@
...
+* Fix deserialization of `SetDigest`, `BingTile`, and `Color` types. ({issue}`24612`)
+
+## BigQuery connector
+
+* Allow configuration of the channel pool for gRPC communication with BigQuery. ({issue}`24638`)
+
+## ClickHouse connector
+
+* {{breaking}} Raise minimum required versions to ClickHouse 24.3 and Altinity
+  22.3. ({issue}`24612`)
```
[the remark this answers is not in the document]
Comment (mosabua) [reply docs/src/main/sphinx/release/release-469.md:44]: 24515 .. updated

File: docs/src/main/sphinx/release/release-469.md [hunk 15]
```diff
@@ -0,0 +1,141 @@
...
+
+## Iceberg connector
+
+* Add support for SSE-C in S3 security mapping. ({issue}`24566`)
+* Allow configuring the endpoint for the Google Storage file system with the
+  `gcs.endpoint` property. ({issue}`24626`)
+* Add `$entries` metadata table. ({issue}`24172`)
+* Add `$all_entries` metadata table. ({issue}`24543`)
+* Add `system.iceberg_tables` table function to allow listing only Iceberg
+  tables. ({issue}`24469`)
```
Comment (ebyhr) [inline docs/src/main/sphinx/release/release-469.md:106]: We should remove this entry as we decided to revert the change. #24719

File: docs/src/main/sphinx/release.md [hunk 16]
```diff
@@ -7,6 +7,7 @@
 :maxdepth: 1
 
 release/release-469
+```
```
Comment (martint) [inline docs/src/main/sphinx/release.md:10]: This should go in the previous commit

File: docs/src/main/sphinx/release/release-469.md [hunk 17]
```diff
@@ -0,0 +1,89 @@
+# Release xyz (dd MMM 2024)
```
Comment (martint) [inline docs/src/main/sphinx/release/release-469.md]: This file should be added in the next commit

File: docs/src/main/sphinx/release/release-469.md [hunk 18]
```diff
@@ -1,89 +1,175 @@
-# Release xyz (dd MMM 2024)
+# Release 469 (16 Jan 2025)
 
 ## General
 
-## Security
+* Add geospatial constructor {func}`ST_GeomFromKML`. ({issue}`24297`)
```
Comment (martint) [inline docs/src/main/sphinx/release/release-469.md:5]: It’s not a constructor. Just say: “Add xxx function”

File: docs/src/main/sphinx/release/release-469.md [hunk 19]
```diff
@@ -1,89 +1,175 @@
...
+* Allow deactivating the spooling protocol with the `spooling_protocol_enabled`
+  session property. ({issue}`24655`)
+* Improve stability of the cluster under load. ({issue}`24572`) 
+* Refrain from join pushdown for modified tables. ({issue}`24447`)
```
Comment (martint) [inline docs/src/main/sphinx/release/release-469.md:9]: I don’t understand what this means

File: docs/src/main/sphinx/release/release-469.md [hunk 20]
```diff
@@ -1,89 +1,175 @@
...
+* Fix parsing of negative `0x`, `0b`, `0o` long literals. ({issue}`24601`)
```
Comment (martint) [inline docs/src/main/sphinx/release/release-469.md:10]: Fix parsing of negative hexadecimal, octal and binary numeric literals.

File: docs/src/main/sphinx/release/release-469.md [hunk 21]
```diff
@@ -1,89 +1,175 @@
...
 
 ## Web UI
 
-## JDBC driver
+* Add support for filtering by query client tags. ({issue}`24494`)
+* Add authentication support for the Preview Web UI with the authentication
```
Comment (martint) [inline docs/src/main/sphinx/release/release-469.md:15]: Remove this one.

File: docs/src/main/sphinx/release/release-469.md [hunk 22]
```diff
@@ -1,89 +1,175 @@
...
+  types `FORM`, `FIXED`, `OAUTH2`, `JWT`, `KERBEROS`, `CERTIFICATE`, and
+  `INSECURE`. ({issue}`23230`)
+* Sort queries on the current worker status section in the Preview Web UI by
```
Comment (martint) [inline docs/src/main/sphinx/release/release-469.md:18]: Same with this one

File: docs/src/main/sphinx/release/release-469.md [hunk 23]
```diff
@@ -1,89 +1,175 @@
...
 
-## Server RPM
+## JDBC driver
 
-## Docker image
+* Add `planningTimeMillis`, `analysisTimeMillis`, `finishingTimeMillis`,
+  `physicalInputBytes`, `physicalWrittenBytes`, `internalNetworkInputBytes` and
+  `physicalInputTimeMillis` to `io.trino.jdbc.QueryStats`. ({issue}`24571`,
+  {issue}`24604`)
+* Add the `validateConnection` connection property to allows validating
```
Comment (martint) [inline docs/src/main/sphinx/release/release-469.md:27]: The grammar in this item is off

File: docs/src/main/sphinx/release/release-469.md [hunk 24]
```diff
@@ -1,89 +1,175 @@
...
+  connection, and improve `Connection.isValid(int)` so it validates connection
+  and credentials. ({issue}`24127`, {issue}`22684`)
+* Fix loading spooled segment data when cluster is secured. ({issue}`24595`)
```
Comment (martint) [inline docs/src/main/sphinx/release/release-469.md:30]: I don’t understand this one

File: docs/src/main/sphinx/release/release-469.md [hunk 25]
```diff
@@ -1,89 +1,175 @@
...
+  files with `ZSTD` by default, instead of the previous `SNAPPY`.
+  ({issue}`17426`)
+* Improve performance of queries on tables with large transaction log JSON
+  files. ({issue}`24491`)
+* Improve performance of reading from Parquet files with a large number of row
+  groups. ({issue}`24618`)
+* Improve performance for the `OPTIMIZE` statement by enabling concurrent
+  execution. ({issue}`16985`)
+* Improve performance of reading from large files on S3. ({issue}`24521`)
+* Fix JMX metrics for file system caching with multiple catalogs. ({issue}`24510`)
```
Comment (martint) [inline docs/src/main/sphinx/release/release-469.md:63]: What was broken?

File: docs/src/main/sphinx/release/release-469.md [hunk 26]
```diff
@@ -1,89 +1,175 @@
...
+* Fix reading from tables when using the Alluxio file system. ({issue}`23815`)
```
Comment (martint) [inline docs/src/main/sphinx/release/release-469.md:64]: What was broken?

File: docs/src/main/sphinx/release/release-469.md [hunk 27]
```diff
@@ -1,89 +1,175 @@
...
+* Fix incorrect results when writing with deletion vectors. ({issue}`24648`)
```
Comment (martint) [inline docs/src/main/sphinx/release/release-469.md:65]: What does “writing with deletion vectors” mean?

File: docs/src/main/sphinx/release/release-469.md [hunk 28]
```diff
@@ -1,89 +1,175 @@
...
 
 ## Hive connector
 
+* Add support for SSE-C in S3 security mapping. ({issue}`24566`)
+* Allow configuring the endpoint for the Google Storage file system with the
+  `gcs.endpoint` property. ({issue}`24626`)
+* Improve performance of reading from Parquet files with a large number of row
+  groups. ({issue}`24618`)
+* Improve performance of reading from large files on S3. ({issue}`24521`)
+* Fix reporting of AWS SDK client retry count metrics to report both client
```
Comment (martint) [inline docs/src/main/sphinx/release/release-469.md:83]: Grammar is off

File: docs/src/main/sphinx/release/release-469.md (hunk 24 above)
Comment (wendigo) [reply docs/src/main/sphinx/release/release-469.md:30]: This is about http client truststore configuration in regard to spooling protocol

File: docs/src/main/sphinx/release/release-469.md (hunk 15 above)
Comment (mosabua) [reply docs/src/main/sphinx/release/release-469.md]: Done

File: docs/src/main/sphinx/release.md (hunk 16 above)
Comment (mosabua) [reply docs/src/main/sphinx/release.md:10]: Not sure how I messed up the commits so badly. Now its fixed.

File: docs/src/main/sphinx/release/release-469.md (hunk 19 above)
Comment (mosabua) [reply docs/src/main/sphinx/release/release-469.md:11]: Reworded .. it prevents planning failures

File: docs/src/main/sphinx/release/release-469.md (hunk 21 above)
Comment (mosabua) [reply docs/src/main/sphinx/release/release-469.md:18]: ok .. not sure I agree .. we did have an entry in the 468 release notes about the preview UI .. should we remove that one as well .. I saw its also in the wrong section in 468 (general vs Web UI)

File: docs/src/main/sphinx/release/release-469.md (hunk 23 above)
Comment (mosabua) [reply docs/src/main/sphinx/release/release-469.md:30]: fixed

File: docs/src/main/sphinx/release/release-469.md (hunk 24 above)
Comment (mosabua) [reply docs/src/main/sphinx/release/release-469.md:33]: Reworded

File: docs/src/main/sphinx/release/release-469.md (hunk 25 above)
Comment (mosabua) [reply docs/src/main/sphinx/release/release-469.md:66]: Reworded

File: docs/src/main/sphinx/release/release-469.md (hunk 28 above)
Comment (mosabua) [reply docs/src/main/sphinx/release/release-469.md:86]: Reworded

File: docs/src/main/sphinx/release/release-469.md (hunk 27 above)
Comment (mosabua) [reply docs/src/main/sphinx/release/release-469.md:68]: I reworded both entries around deletion vectors .. maybe @ebyhr and @findinpath can confirm .. essentially deletion vectors is a feature that you can enable on Delta Lake tables.. and when you do that .. there are scenario where writing as well as reading data from the table is wrong like wrong row counts and such

File: docs/src/main/sphinx/release/release-469.md [hunk 29]
```diff
@@ -0,0 +1,186 @@
+# Release 469 (21 Jan 2025)
+
+## General
+
+* Add support for the `FIRST`, `AFTER`, and `LAST` clauses to `ALTER TABLE ...
+  ADD COLUMN`. ({issue}`20091`)
+* Add the geospatial {func}`ST_GeomFromKML` function. ({issue}`24297`)
+* Allow deactivating the spooling protocol with the `spooling_protocol_enabled`
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-469.md:8]: Should we collapse these and not add any of the property names since they are added to the docs in #24761 @wendigo ?

Comment (wendigo) [reply docs/src/main/sphinx/release/release-469.md]: yeah, makes sense

Comment (mosabua) [reply docs/src/main/sphinx/release/release-469.md]: Done

File: docs/src/main/sphinx/release/release-469.md [hunk 30]
```diff
@@ -0,0 +1,192 @@
...
+## Elasticsearch connector
+
+* Improve performance of queries that reference nested fields from Elasticsearch
+  documents. ({issue}`23069`)
+
+## Faker connector
+
+* Add support for views. ({issue}`24242`)
+* Support generating sequences. ({issue}`24590`)
+* {{breaking}} Replace predicate pushdown with the `min`, `max`, and `options`
```
Comment (martint) [inline docs/src/main/sphinx/release/release-469.md:79]: This is not about replacing predicate pushdown, but changing the mechanism for specifying the constraints of the generated data. Previously, it would infer them from the WHERE clause. Now, they need to be provided as part of the table / column declaration via column properties.

File: docs/src/main/sphinx/release/release-469.md [hunk 31]
```diff
@@ -0,0 +1,192 @@
...
+  `phoenix.server-scan-page-timeout` configuration property. ({issue}`24689`)
+* Fix failure when updating values to `NULL`. ({issue}`24204`)
+
+## PostgreSQL connector
+
+* Fix failure when updating values to `NULL`. ({issue}`24204`)
+
+## Redshift connector
+
+* Improve performance of reading from Redshift tables by using the Redshift
```
Comment (martint) [inline docs/src/main/sphinx/release/release-469.md:169]: "by using the Redshift UNLOAD command" is an implementation detail. I'd remove it, or at most say "by increasing parallelism"

File: docs/src/main/sphinx/release/release-469.md [hunk 32]
```diff
@@ -0,0 +1,192 @@
...
+  files. ({issue}`24491`)
+* Improve performance of reading from Parquet files with a large number of row
+  groups. ({issue}`24618`)
+* Improve performance for the `OPTIMIZE` statement by enabling concurrent
+  execution. ({issue}`16985`)
+* Improve performance of reading from large files on S3. ({issue}`24521`)
+* Correct catalog information in JMX metrics when using file system caching with
+  multiple catalogs. ({issue}`24510`)
+* Fix table read failures when using the Alluxio file system. ({issue}`23815`)
+* Fix incorrect results updating tables with deletion vectors enabled. ({issue}`24648`)
```
[the remark this answers is not in the document]
Comment (ebyhr) [reply docs/src/main/sphinx/release/release-469.md:67]: Incorrect results.

File: docs/src/main/sphinx/release/release-469.md [hunk 33]
```diff
@@ -0,0 +1,192 @@
...
+
+* Add support for the `FIRST`, `AFTER`, and `LAST` clauses to `ALTER TABLE ...
+  ADD COLUMN`. ({issue}`20091`)
+* Add the geospatial {func}`ST_GeomFromKML` function. ({issue}`24297`)
+* Allow configuring the spooling client protocol behaviour with session
+  properties. ({issue}`24655`, {issue}`24757`)
+* Improve stability of the cluster under load. ({issue}`24572`)
+* Prevent planning failures resulting from join pushdown for modified tables. ({issue}`24447`)
+* Fix parsing of negative hexadecimal, octal, and binary numeric literals. ({issue}`24601`)
+* Fix failures with recursive delete operations on S3Express preventing usage
```
[the remark this answers is not in the document]
Comment (mosabua) [reply docs/src/main/sphinx/release/release-469.md:13]: Cleaned up

File: docs/src/main/sphinx/release/release-469.md [hunk 34]
```diff
@@ -0,0 +1,192 @@
...
+
+* Add support for filtering queries by client tags. ({issue}`24494`)
+
+## JDBC driver
+
+* Add `planningTimeMillis`, `analysisTimeMillis`, `finishingTimeMillis`,
+  `physicalInputBytes`, `physicalWrittenBytes`, `internalNetworkInputBytes` and
+  `physicalInputTimeMillis` to `io.trino.jdbc.QueryStats`. ({issue}`24571`,
+  {issue}`24604`)
+* Add the `validateConnection` connection property to allow validating
```
[the remark this answers is not in the document]
Comment (mosabua) [reply docs/src/main/sphinx/release/release-469.md:26]: Simplified

File: docs/src/main/sphinx/release/release-469.md [hunk 35]
```diff
@@ -0,0 +1,192 @@
...
+  connection, and improve the `Connection.isValid(int)` method so it validates
+  connection and credentials. ({issue}`24127`, {issue}`22684`)
+* Prevent failures when using the spooling protocol with a cluster using its own
```
[the remark this answers is not in the document]
Comment (mosabua) [reply docs/src/main/sphinx/release/release-469.md:29]: I think .. its own certificate chain up to a custom root certifcate from a custom internal CA .. but @wendigo would have to confirm or help with rewording. We can also chat in a call and look at the issue together

File: docs/src/main/sphinx/release/release-469.md (hunk 32 above)
[the remark this answers is not in the document]
Comment (mosabua) [reply docs/src/main/sphinx/release/release-469.md:67]: Done

File: docs/src/main/sphinx/release/release-469.md (hunk 30 above)
Comment (mosabua) [reply docs/src/main/sphinx/release/release-469.md:79]: I got that wording from @nineinchnick .. will reword though

Comment (mosabua) [reply docs/src/main/sphinx/release/release-469.md:79]: Done

File: docs/src/main/sphinx/release/release-469.md (hunk 31 above)
Comment (mosabua) [reply docs/src/main/sphinx/release/release-469.md:169]: I removed it..

**Review by martint — APPROVED** [review]
